pkg/openshift/rosa/hcpvpc.go

```diff
@@ -0,0 +1,159 @@
+package rosa
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/openshift/osde2e-common/assets"
+	"github.com/openshift/osde2e-common/internal/terraform"
+
+	"github.com/hashicorp/terraform-exec/tfexec"
+)
+
+// vpc represents the details of an aws vpc
+type vpc struct {
+	privateSubnet     string
+	publicSubnet      string
+	nodePrivateSubnet string
+}
+
+// hpcVPCError represents the custom error
+type hcpVPCError struct {
+	action string
+	err    error
+}
+
+// Error returns the formatted error message when hpcVPCError is invoked
+func (h *hcpVPCError) Error() string {
+	return fmt.Sprintf("%s hcp cluster vpc failed: %v", h.action, h.err)
+}
+
+// copyFile copies the srcFile provided to the destFile
+func copyFile(srcFile, destFile string) error {
+	srcReader, err := assets.FS.Open(srcFile)
+	if err != nil {
+		return fmt.Errorf("error opening %s file: %w", srcFile, err)
+	}
+	defer srcReader.Close()
+
+	destReader, err := os.Create(destFile)
+	if err != nil {
+		return fmt.Errorf("error creating runtime %s file: %w", destFile, err)
+	}
+	defer destReader.Close()
+
+	_, err = io.Copy(destReader, srcReader)
+	if err != nil {
+		return fmt.Errorf("error copying source file to destination file: %w", err)
+	}
+
+	return nil
+}
+
+// createHostedControlPlaneVPC creates the aws vpc used for provisioning hosted control plane clusters
+func (r *Provider) createHostedControlPlaneVPC(ctx context.Context, clusterName, awsRegion, workingDir string) (*vpc, error) {
+	action := "create"
+	var vpc vpc
+
+	if clusterName == "" || awsRegion == "" || workingDir == "" {
+		return nil, &hcpVPCError{action: action, err: errors.New("one or more parameters is empty")}
```

Comment:
They are bubbled up to the caller of the function where they will either print the return error or raise it up (if they do) and that caller will print it.

Since each command is implementing their own custom error, the message I feel is pretty clear and tells the user exactly where the failure is coming from. We can always revise this to print it directly where it happened too if we want in a separate PR.